go/DEVELOPER.md

```diff
@@ -0,0 +1,206 @@
+# Developer Guide
+
+This document describes how to set up your development environment to build and test the GLIDE for Redis Go wrapper.
+
+### Development Overview
+
+The GLIDE for Redis Go wrapper consists of both Go and Rust code. The Go and Rust components communicate in two ways:
+1. using the [protobuf](https://github.com/protocolbuffers/protobuf) protocol.
+2. using shared C objects. [cgo](https://pkg.go.dev/cmd/cgo) is used to interact with the C objects from Go code.
+
+### Build from source
+
+#### Prerequisites
+
+Software Dependencies
+
+-   Go
+-   GNU Make
+-   git
+-   GCC
+-   pkg-config
+-   protoc (protobuf compiler) >= v3.20.0
+-   openssl
+-   openssl-dev
+-   rustup
+
+
+**Dependencies installation for Ubuntu**
+
+```bash
+sudo apt update -y
+sudo apt install -y git gcc pkg-config openssl libssl-dev unzip make
+# Install Go
+sudo snap install go --classic
+export PATH="$PATH:$HOME/go/bin"
+# Install rust
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+source "$HOME/.cargo/env"
+# Check that the Rust compiler is installed
+rustc --version
+# Install protobuf compiler
+PB_REL="https://github.com/protocolbuffers/protobuf/releases"
+curl -LO $PB_REL/download/v3.20.3/protoc-3.20.3-linux-x86_64.zip
+unzip protoc-3.20.3-linux-x86_64.zip -d $HOME/.local
+export PATH="$PATH:$HOME/.local/bin"
+# Check that the protobuf compiler is installed
+protoc --version
+```
+
+**Dependencies installation for CentOS**
+
+```bash
+sudo yum update -y
+sudo yum install -y git gcc pkgconfig openssl openssl-devel unzip wget tar
+# Install Go
+wget https://go.dev/dl/go1.22.0.linux-amd64.tar.gz
+sudo tar -C /usr/local -xzf go1.22.0.linux-amd64.tar.gz
+export PATH="$PATH:/usr/local/go/bin"
+export PATH="$PATH:$HOME/go/bin"
+# Install rust
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+source "$HOME/.cargo/env"
+# Check that the Rust compiler is installed
+rustc --version
+# Install protobuf compiler
+PB_REL="https://github.com/protocolbuffers/protobuf/releases"
+curl -LO $PB_REL/download/v3.20.3/protoc-3.20.3-linux-x86_64.zip
+unzip protoc-3.20.3-linux-x86_64.zip -d $HOME/.local
+export PATH="$PATH:$HOME/.local/bin"
+# Check that the protobuf compiler is installed
+protoc --version
+```
+
+**Dependencies installation for MacOS**
+
+```bash
```

Comment:
I'm not sure how to test these steps on mac. With CentOS and Ubuntu I was able to test by creating a new AMI instance using my AWS isengard account, but I keep running into errors when trying to create a Mac AMI. I can't fully test the Mac steps on my own Mac since I already have everything installed. Any suggestions?